### PR TITLE
add csv download button

### LINKED ISSUE #106: [NEW] Enable downloading table as CSV
Depends on:
- #72 
- #105

Once the table is editable, we should add the option to download just the table as a CSV.

### PR DESCRIPTION
**Pull request by ermannolocascio**
fix #106

### CODE REVIEW

File: includes/downloader.js
Comment (sjvrijn) [inline includes/downloader.js:169]: This function already exists earlier in the same file

**Review by sjvrijn — CHANGES_REQUESTED** [review]
Looks good, thanks for implementing!

Two issues:
- this PR duplicates the `getTimeStampString()` function
- I'm not sure where this happens, but when downloading the CSV I get extra lines that only contain a double quote `"`:

```csv
Id,Statement Type,Attribute,Deontic,Aim,Direct Object,Type Of Direct Object,Indirect Object,Type Of Indirect Object,Activation Condition,Execution Constraint,Extra Column
"1","formal","VROMI minister","must","order","infrastructure dept to execute clean-up","animate","","","if necessary after a storm event; but only during a full supermoon when both Jupiter and Saturn are in retrograde","","
"
"2","informal","Governor","","requests","financial aid","inanimate","from Dutch Kingdom","animate","if requested by Prime minister","","
"
"3","formal","Property owners","must","insure","their properties","inanimate","","","","","
"
"4","informal","NGOs","","help","","","","","if requested by inhabitants","in reconstruction","
"
"5","formal","VROMI infrastructure dept.","must","clean","gutters","inanimate","","","","before June (start hurricane season)","
"
"6","formal","ACER","must","update","the recommendations","inanimate","","","","at least once every two years",""
```

To reproduce:
- upload [ina-demo.json](https://github.com/user-attachments/files/17012472/ina-demo.json)
- click 'Download as CSV' button
- observe extra lines in downloaded file

File: includes/downloader.js
Comment (suvayu) [inline includes/downloader.js:134]: > * I'm not sure where this happens, but when downloading the CSV I get extra lines that only contain a double quote `"`:

I think what is happening is that instead of adding an empty quote, the trailing newline is getting wrapped in quotes `\n` -> `"\n"`.

I'm not exactly sure yet how this is happening (don't know the behaviour of `replace` in JS), but I think doing the following would do it
```javascript
if (value == '') {
    return value;
} else {
    return ...; // escape
}
```

Comment (suvayu) [reply includes/downloader.js:134]: Actually I'm less sure now, maybe my analysis is incorrect :frowning_face:

**Comment by ermannolocascio** [comment]
hi guys, I have just implemented few changes which should address your observations (see last commit) - thank you. 
However, I have noticed two problems. Precisely, if I download the .csv file (columns and rows only) and I try to upload the same file in a second moment, all the connections are lost (**problem A**). 
Also, the _Or Else_ column is named _Extra Column_ in the `INA.columnNames` variable: this generates an error during the uploading phase (**problem B**). 

For **problem A**, I guess we should clarify first what's the purpose of this csv download button. If the downloaded file is needed to resume the project in a second moment, then we should elaborate a solution that does not loose the info relative to the connections between statements.

For **problem B**, @sjvrijn please, do you remember why the column is renamed at some point in the process?

Please, let me know your idea @suvayu and @sjvrijn @OleMussmann / Cheers :)

**Comment by sjvrijn** [comment]
To me, **A** is not a problem, since I'm expecting the CSV download to only download the data in the table, which does not include the connections. I also wouldn't know how to nicely include that in the CSV anyway. For remembering the connections, we came up with the project .json download instead. We will have to make sure that it's clear which download option has which functionality, to not confuse people expecting connections to be remembered when downloading the table.

**Comment by suvayu** [comment]
> We will have to make sure that it's clear which download option has which functionality, to not confuse people expecting connections to be remembered when downloading the table.

Why shouldn't we always download everything (i.e. project + data)?

**Comment by ermannolocascio** [comment]
Thanks guys. Then, let's keep it as it is and wait for a feedback from the users as discussed during the meeting. 

@sjvrijn - should we plan a meeting with the users? I can do that, or, if you prefer, you do that. Ups to you. Cheers e

**Comment by ermannolocascio** [comment]
Hi @sjvrijn @suvayu , should we temporarly merge this PR? I think once we get a sort of UI feedback from the users then we move the button elsewhere. Cheers, E

**Comment by suvayu** [comment]
As long as you create an issue documenting the compromise (or unresolved issue), I think it should be okay to merge.

**Review by sjvrijn — APPROVED** [review]
The downloaded CSV looks fine now, good to merge!

**Pull request merged**
